File: Thermian/Application/HotspotSystem/HotspotApp/src/HotspotDataset/UseCase/FindImage/FindImageUseCase.php
<?php
declare(strict_types=1);

namespace Hotspot\HotspotDataset\UseCase\FindImage;

use Hotspot\HotspotDataset\Domain\Image\Image;
use Hotspot\HotspotDataset\Domain\Image\ImageId;
use Hotspot\HotspotDataset\Domain\ImageRepository\ImageRepository;

class FindImageUseCase
{
    private ImageRepository $repository;

    public function __construct(ImageRepository $repository)
    {
        $this->repository = $repository;
    }

    public function execute(FindImageQuery $query): ?Image
    {
        $imageId = ImageId::fromString($query->getImageId());

        return $this->repository->findById($imageId);
    }
}

File: Thermian/Application/HotspotSystem/HotspotAI/src/dataset/use_case/find_image/FindImageUseCase.py
from src.dataset.domain.input_image_repository import InputImageRepository
from src.dataset.use_case.find_image.FindImageCommand import FindImageCommand
from src.detection.domain.files.Image import Image


class FindImageUseCase:

    def __init__(self, image_repository: InputImageRepository) -> None:
        self.__image_repository = image_repository

    def execute(self, command: FindImageCommand) -> Image:
        return self.__image_repository.find(command.image_id)
